cpu/stm32_common/include/periph_cpu_common.h

```diff
@@ -370,6 +370,52 @@ typedef enum {
     STM32_LPUART,           /**< STM32 Low-power UART (LPUART) module type */
 } uart_type_t;
 
+#ifdef MODULE_PERIPH_UART_MODECFG
+#define UART_INVALID_MODE 0xFF
```

Comment:
If you just make it a bit this will force the uint32_t for the enum 
`#define UART_INVALID_MODE 0x8000000`
(please add a comment why)